From: Patrik Fimml <patrik@fimml.at>
Subject: [PATCH] debian/libtool-as-needed

Fix ltmain.sh to correctly put -Wl,--as-needed before library arguments.

Patch by Josselin Mouette <joss@debian.org> in
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=347650

Signed-off-by: Patrik Fimml <patrik@fimml.at>

---
 ltmain.sh |   14 ++++++++++++++
 1 files changed, 14 insertions(+), 0 deletions(-)

diff --git a/ltmain.sh b/ltmain.sh
index a72f2fd..fd4cd52 100755
--- a/ltmain.sh
+++ b/ltmain.sh
@@ -4716,6 +4716,11 @@ func_mode_link ()
 	arg=$func_stripname_result
 	;;
 
+      -Wl,--as-needed)
+	deplibs="$deplibs $arg"
+	continue
+	;;
+
       -Wl,*)
 	func_stripname '-Wl,' '' "$arg"
 	args=$func_stripname_result
@@ -5067,6 +5072,15 @@ func_mode_link ()
 	lib=
 	found=no
 	case $deplib in
+	-Wl,--as-needed)
+	  if test "$linkmode,$pass" = "prog,link"; then
+	    compile_deplibs="$deplib $compile_deplibs"
+	    finalize_deplibs="$deplib $finalize_deplibs"
+	  else
+	    deplibs="$deplib $deplibs"
+	  fi
+	  continue
+	  ;;
 	-mt|-mthreads|-kthread|-Kthread|-pthread|-pthreads|--thread-safe|-threads)
 	  if test "$linkmode,$pass" = "prog,link"; then
 	    compile_deplibs="$deplib $compile_deplibs"
-- 
tg: (a60ef3b..) debian/libtool-as-needed (depends on: master)
